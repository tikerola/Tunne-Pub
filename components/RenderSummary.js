import React, { useState } from "react";
import {
  FlatList,
  View,
  Image,
  Text,
  StyleSheet,
  TouchableOpacity,
  ScrollView,
  Modal,
} from "react-native";
import { MaterialIcons } from "@expo/vector-icons";
import {
  deleteSummaryItem,
  setActiveSummaryItemId,
  setScreenNumber,
} from "../store/actions/cards";
import { useSelector } from "react-redux";
import AsyncStorage from "@react-native-community/async-storage";
import CustomButton from "./CustomButton";
import DeleteModal from "./DeleteModal";

export const RenderSummary = ({
  item,
  dispatch,
  enableEdit = false,
  navigation,
}) => {
  const [modalVisible, setModalVisible] = useState(false);
  const [deleteId, setDeleteId] = useState("");

  const summaryItems = useSelector((state) => state.cards.summaryOfChosenItems);

  const handleDelete = async () => {
    navigation.goBack();
    dispatch(deleteSummaryItem(deleteId));
    const items = { ...summaryItems };
    delete items[deleteId];

    try {
      AsyncStorage.setItem("items", JSON.stringify(items));
    } catch (err) {
      console.log("poisto epÃ¤onnistui");
    }
  };

  const itemQuestionsToArray = React.useCallback(() => {
    const arr = [];

    for (let i in item.questions) {
      arr.push({ [i]: item.questions[i] });
    }

    return arr;
  }, [item]);

  const handleEdit = () => {
    dispatch(setActiveSummaryItemId(item.id));
    navigation.navigate("MainScreen", { name: summaryItems[item.id].name });
    dispatch(setScreenNumber(41));
  };

  const toggleModal = () => setModalVisible((prevState) => !prevState);

  if (!item) return null;

  if (modalVisible)
    return (
      <DeleteModal
        modalVisible={modalVisible}
        setModalVisible={setModalVisible}
        handleDelete={handleDelete}
        toggleModal={toggleModal}
      />
    );

  return (
    <ScrollView contentContainerStyle={styles.itemContainer}>
      <FlatList
        horizontal
        data={item.cards}
        keyExtractor={(item, index) => index.toString()}
        renderItem={({ item }) => {
          return <Image style={styles.image} source={item.filename} />;
        }}
      />

      <View>
        {itemQuestionsToArray().map((q, index) => {
          if (Object.values(q)[0] === "") return;

          return (
            <View style={styles.questionsContainer} key={index}>
              <Text
                style={{
                  color: "#bbb",
                  fontSize: 18,
                  fontFamily: "Roboto-Light",
                }}
              >
                <Text style={styles.bold}>{Object.keys(q)[0]}: </Text>
                {Object.values(q)[0]}
              </Text>
            </View>
          );
        })}
      </View>
      {enableEdit && (
        <View style={styles.imageAndWordsContainer}>
          <View style={styles.wordAndButtonContainer}>
            <View style={styles.buttonContainer}>
              <CustomButton title="Muokkaa" onPress={handleEdit} />
              <CustomButton
                title="Poista"
                onPress={() => {
                  setDeleteId(item.id);
                  toggleModal();
                }}
                color="red"
              />
            </View>
          </View>
        </View>
      )}
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  itemContainer: {
    width: "100%",
    display: "flex",
    flexDirection: "column",
    paddingTop: 10,
    paddingBottom: 15,
  },

  imageAndWordsContainer: {
    display: "flex",
    flexDirection: "row",
    marginTop: 10,
  },

  wordAndButtonContainer: {
    marginLeft: 10,
    flex: 1,
    display: "flex",
    flexDirection: "column",
    justifyContent: "space-between",
    paddingBottom: 5,
  },

  image: {
    width: 96,
    height: 140,
    marginLeft: 5,
    marginRight: 5,
  },

  buttonContainer: {
    marginLeft: "auto",
    marginRight: "auto",
    width: "75%",
    alignSelf: "center",
    marginTop: 10,
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-between",
  },

  questionsContainer: {
    marginTop: 10,
    width: "100%",
    display: "flex",
    flexDirection: "row",
    backgroundColor: "rgba(0,0,0,0.5)",
    padding: 15,
  },

  bold: {
    fontWeight: "bold",
    color: "#666",
  },
});
