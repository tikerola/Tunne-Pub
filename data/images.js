import image1 from "../assets/images/1.jpg";
import image2 from "../assets/images/2.jpg";
import image3 from "../assets/images/3.jpg";
import image4 from "../assets/images/4.jpg";
import image5 from "../assets/images/5.jpg";
import image6 from "../assets/images/6.jpg";
import image7 from "../assets/images/7.jpg";
import image8 from "../assets/images/8.jpg";
import image9 from "../assets/images/9.jpg";
import image10 from "../assets/images/10.jpg";
import image11 from "../assets/images/11.jpg";
import image12 from "../assets/images/12.jpg";
import image13 from "../assets/images/13.jpg";
import image14 from "../assets/images/14.jpg";
import image15 from "../assets/images/15.jpg";
import image16 from "../assets/images/16.jpg";
import image17 from "../assets/images/17.jpg";
import image18 from "../assets/images/18.jpg";
import image19 from "../assets/images/19.jpg";
import image20 from "../assets/images/20.jpg";
import image21 from "../assets/images/21.jpg";
import image22 from "../assets/images/22.jpg";
import image23 from "../assets/images/23.jpg";
import image24 from "../assets/images/24.jpg";
import image25 from "../assets/images/25.jpg";
import image26 from "../assets/images/26.jpg";
import image27 from "../assets/images/27.jpg";
import image28 from "../assets/images/28.jpg";
import image29 from "../assets/images/29.jpg";
import image30 from "../assets/images/30.jpg";
import image31 from "../assets/images/31.jpg";
import image32 from "../assets/images/32.jpg";
import image33 from "../assets/images/33.jpg";
import image34 from "../assets/images/34.jpg";
import image35 from "../assets/images/35.jpg";
import image36 from "../assets/images/36.jpg";
import image37 from "../assets/images/37.jpg";
import image38 from "../assets/images/38.jpg";
import image39 from "../assets/images/39.jpg";
import image40 from "../assets/images/40.jpg";
import image41 from "../assets/images/41.jpg";
import image42 from "../assets/images/42.jpg";
import image43 from "../assets/images/43.jpg";
import image44 from "../assets/images/44.jpg";
import image45 from "../assets/images/45.jpg";
import image46 from "../assets/images/46.jpg";
import image47 from "../assets/images/47.jpg";
import image48 from "../assets/images/48.jpg";
import image49 from "../assets/images/49.jpg";

export {
  image1,
  image2,
  image3,
  image4,
  image5,
  image6,
  image7,
  image8,
  image9,
  image10,
  image11,
  image12,
  image13,
  image14,
  image15,
  image16,
  image17,
  image18,
  image19,
  image20,
  image21,
  image22,
  image23,
  image24,
  image25,
  image26,
  image27,
  image28,
  image29,
  image30,
  image31,
  image32,
  image33,
  image34,
  image35,
  image36,
  image37,
  image38,
  image39,
  image40,
  image41,
  image42,
  image43,
  image44,
  image45,
  image46,
  image47,
  image48,
  image49,
};
