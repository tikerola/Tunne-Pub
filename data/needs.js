const NEEDS_DATA = [
  { id: '1', word: 'Arvostus' },
  { id: '2', word: 'empatia' },
  { id: '3', word: 'hellyys' },
  { id: '4', word: 'henkinen turva' },
  { id: '5', word: 'merkityksellisyys' },
  { id: '6', word: 'huolenpito' },
  { id: '7', word: 'hoiva' },
  { id: '8', word: 'huomioiduksi tuleminen' },
  { id: '9', word: 'hyväksyntä' },
  { id: '10', word: 'Johdonmukaisuus' },
  { id: '11', word: 'kiintymys' },
  { id: '12', word: 'kumppanuus' },
  { id: '13', word: 'kunnioitus' },
  { id: '14', word: 'luottamus' },
  { id: '15', word: 'läheisyys' },
  { id: '16', word: 'myötätunto' },
  { id: '17', word: 'rakkaus' },
  { id: '18', word: 'nähdyksi tuleminen' },
  { id: '19', word: 'kuulluksi tuleminen' },
  { id: '20', word: 'Vastavuoroisuus' },
  { id: '21', word: 'vuorovaikutus' },
  { id: '22', word: 'ymmärretyksi tuleminen' },
  { id: '23', word: 'Hauskanpito' },
  { id: '24', word: 'huumori' },
  { id: '25', word: 'seikkailu' },
  { id: '26', word: 'Aitous' },
  { id: '27', word: 'rehellisyys' },
  { id: '28', word: 'läsnäolo' },
  { id: '29', word: 'Tieto' },
  { id: '30', word: 'oma tila' },
  { id: '31', word: 'vaihtelu' },
  { id: '32', word: 'vapaus' },
  { id: '33', word: 'vastuu' },
  { id: '34', word: 'yksityisyys' },
  { id: '35', word: 'ennakoitavuus' },
  { id: '36', word: 'kauneus' },
  { id: '37', word: 'Lohtu' },
  { id: '38', word: 'reiluus' },
  { id: '39', word: 'tasavertaisuus' },
  { id: '40', word: 'suojeleminen' },
  { id: '41', word: 'tasapaino' },
  { id: '42', word: 'yhteisöllisyys' },
  { id: '43', word: 'Kosketus' },
  { id: '44', word: 'lepo' },
  { id: '45', word: 'uni' },
  { id: '46', word: 'liikkuminen' },
  { id: '47', word: 'nautinto' },
  { id: '48', word: 'ravinto' },
  { id: '49', word: 'rentoutuminen' },
  { id: '50', word: 'seksuaalinen ilmaisu' },
  { id: '51', word: 'Aikaansaaminen' },
  { id: '52', word: 'haasteellisuus' },
  { id: '53', word: 'kiitollisuus' },
  { id: '54', word: 'hengellisyys' },
  { id: '55', word: 'itseilmaisu' }
]

export default NEEDS_DATA





