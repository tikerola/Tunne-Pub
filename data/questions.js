







export const QUESTIONS = [
  'Sinulla on ollut jokin tunteitanostattava tilanne? Kerro mitä tapahtui.',
  'Sanat, jotka parhaiten kuvaavat tunteitasi',
  'Kuvat, jotka parhaiten kuvaavat tunteitasi.',
  'Missä ja miten tunne tuntuu kehossa (esim. paino rinnan päällä, pala kurkussa, kädet menee nyrkkiin)',
  'Onko tunne alunperin sinun vai nappasitko jonkun toisen tunteen?',
  'Mikä tarve tunteen takana? Mitä olisit siinä hetkessä tarvinnut? (kuulluksi tulemista, lohdutusta, jne)',
  'Mitä tunnetta olisit kaivannut?',
  'Mikä tarve sinulla on juuri nyt?',
  'Mitä lausetta pyörität päässäsi juuri nyt? Mikä uskomus sinulla on? Mitä se sinulle merkitsee?',
  'Mitä sinun nyt pitäisi tehdä jotta voisit hyvin?'
]