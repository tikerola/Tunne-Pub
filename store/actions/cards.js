export const TOGGLE_CHOSEN_CARD = "TOGGLE CHOSEN CARD";
export const ADD_SUMMARY_ITEM = "ADD SUMMARY ITEM";
export const DELETE_SUMMARY_ITEM = "DELETE SUMMARY ITEM";
export const ADD_ANSWER_TO_A_QUESTION = "ADD ANSWER TO A QUESTION";
export const LOAD_SUMMARY_ITEMS = "LOAD SUMMARY ITEMS";

export const SET_ACTIVE_SUMMARY_ITEM_ID = "SET ACTIVE SUMMARY ITEM ID";
export const SET_SCREEN_NUMBER = "SET SCREEN NUMBER";
export const ADD_SUMMARY_ITEM_NAME = "ADD SUMMARY ITEM NAME";
export const ITEM_SAVED = "ITEM SAVED";
export const SET_NUM_OF_ITEMS = "SET NUM OF ITEMS";
export const TOGGLE_WORD = "TOGGLE WORD";

export const toggleWord = (typeOfWord, word, isAdd, question) => ({
  type: TOGGLE_WORD,
  typeOfWord,
  payload: word,
  isAdd,
  question,
});

export const setNumOfItems = (num) => {
  return {
    type: SET_NUM_OF_ITEMS,
    num,
  };
};

export const itemSaved = (id) => {
  return {
    type: ITEM_SAVED,
    id,
  };
};

export const setScreenNumber = (screenNumber) => {
  return {
    type: SET_SCREEN_NUMBER,
    screenNumber,
  };
};

export const setActiveSummaryItemId = (id) => ({
  type: SET_ACTIVE_SUMMARY_ITEM_ID,
  id,
});

export const toggleChosenCard = (card, isAdd) => {
  return {
    type: TOGGLE_CHOSEN_CARD,
    payload: card,
    isAdd,
  };
};

export const addSummaryItem = (id) => {
  return {
    type: ADD_SUMMARY_ITEM,
    id,
  };
};

export const addSummaryItemName = (id, name) => {
  return {
    type: ADD_SUMMARY_ITEM_NAME,
    id,
    name,
  };
};

export const deleteSummaryItem = (id) => {
  return {
    type: DELETE_SUMMARY_ITEM,
    id,
  };
};

export const addAnswerToAQuestion = (
  summaryItemId,
  question,
  answer,
  listMode,
  wordType
) => {
  return {
    type: ADD_ANSWER_TO_A_QUESTION,
    id: summaryItemId,
    question,
    answer,
    listMode,
    wordType,
  };
};

export const loadSummaryItems = (items) => {
  return {
    type: LOAD_SUMMARY_ITEMS,
    items,
  };
};
