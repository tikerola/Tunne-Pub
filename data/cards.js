import Card from "../models/card";
import * as images from "./images";

const CARD_DATA = [
  new Card("1", "happy", images.image1),
  new Card("2", "sad", images.image2),
  new Card("3", "worried", images.image3),
  new Card("4", "scared", images.image4),
  new Card("5", "happy", images.image5),
  new Card("6", "sad", images.image6),
  new Card("7", "worried", images.image7),
  new Card("8", "scared", images.image8),

  new Card("9", "happy", images.image9),
  new Card("10", "sad", images.image10),
  new Card("11", "worried", images.image11),
  new Card("12", "scared", images.image12),
  new Card("13", "happy", images.image13),
  new Card("14", "sad", images.image14),
  new Card("15", "worried", images.image15),
  new Card("16", "scared", images.image16),

  new Card("17", "happy", images.image17),
  new Card("18", "sad", images.image18),
  new Card("19", "worried", images.image19),
  new Card("20", "scared", images.image20),
  new Card("21", "happy", images.image21),
  new Card("22", "sad", images.image22),
  new Card("23", "worried", images.image23),
  new Card("24", "scared", images.image24),

  new Card("25", "happy", images.image25),
  new Card("26", "sad", images.image26),
  new Card("27", "worried", images.image27),
  new Card("28", "scared", images.image28),
  new Card("29", "happy", images.image29),
  new Card("30", "sad", images.image30),
  new Card("31", "worried", images.image31),
  new Card("32", "scared", images.image32),

  new Card("33", "happy", images.image33),
  new Card("34", "sad", images.image34),
  new Card("35", "worried", images.image35),
  new Card("36", "scared", images.image36),
  new Card("37", "happy", images.image37),
  new Card("38", "sad", images.image38),
  new Card("39", "worried", images.image39),
  new Card("40", "scared", images.image40),

  new Card("41", "happy", images.image41),
  new Card("42", "sad", images.image42),
  new Card("43", "worried", images.image43),
  new Card("44", "scared", images.image44),
  new Card("45", "happy", images.image45),
  new Card("46", "sad", images.image46),
  new Card("47", "worried", images.image47),
  new Card("48", "scared", images.image48),
  new Card("49", "scared", images.image49),
];

export default CARD_DATA;
