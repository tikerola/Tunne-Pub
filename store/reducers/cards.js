import CART_DATA from "../../data/cards";
import WORD_DATA from "../../data/words";
import NEEDS_DATA from "../../data/needs";
import { QUESTIONS } from "../../data/questions";

import {
  TOGGLE_CHOSEN_CARD,
  ADD_SUMMARY_ITEM,
  DELETE_SUMMARY_ITEM,
  ADD_ANSWER_TO_A_QUESTION,
  LOAD_SUMMARY_ITEMS,
  SET_ACTIVE_SUMMARY_ITEM_ID,
  SET_SCREEN_NUMBER,
  ADD_SUMMARY_ITEM_NAME,
  ITEM_SAVED,
  SET_NUM_OF_ITEMS,
  TOGGLE_WORD,
} from "../actions/cards";
import { SENTENCES_DATA } from "../../data/sentences";

const initialState = {
  cards: CART_DATA,
  words: WORD_DATA,
  needs: NEEDS_DATA,
  sentences: SENTENCES_DATA,
  activeSummaryItemId: "",
  summaryOfChosenItems: {},
  screenNumber: 1,
  numOfItems: 0,
  initialLoadFinished: false,
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case SET_NUM_OF_ITEMS:
      return {
        ...state,
        numOfItems: action.num,
      };

    case ITEM_SAVED: {
      const summaryItems = { ...state.summaryOfChosenItems };
      summaryItems[action.id].itemSaved = true;

      return {
        ...state,
        summaryOfChosenItems: summaryItems,
      };
    }

    case ADD_SUMMARY_ITEM_NAME: {
      const summaryItems = { ...state.summaryOfChosenItems };
      summaryItems[action.id].name = action.name;

      return {
        ...state,
        summaryOfChosenItems: summaryItems,
      };
    }

    case SET_SCREEN_NUMBER:
      return {
        ...state,
        screenNumber: action.screenNumber,
      };

    case SET_ACTIVE_SUMMARY_ITEM_ID:
      return {
        ...state,
        activeSummaryItemId: action.id,
      };

    case TOGGLE_CHOSEN_CARD: {
      const itemToEdit = {
        ...state.summaryOfChosenItems[state.activeSummaryItemId],
      };

      if (action.isAdd)
        itemToEdit.cards = [...itemToEdit.cards, action.payload];
      else
        itemToEdit.cards = itemToEdit.cards.filter(
          (card) => card.id !== action.payload.id
        );

      return {
        ...state,
        summaryOfChosenItems: {
          ...state.summaryOfChosenItems,
          [state.activeSummaryItemId]: itemToEdit,
        },
      };
    }

    case TOGGLE_WORD: {
      const itemToEdit = {
        ...state.summaryOfChosenItems[state.activeSummaryItemId],
      };

      if (action.isAdd) {
        itemToEdit[action.typeOfWord] = [
          ...itemToEdit[action.typeOfWord],
          action.payload,
        ];
      } else
        itemToEdit[action.typeOfWord] = itemToEdit[action.typeOfWord].filter(
          (word) => word !== action.payload
        );

      if (action.typeOfWord === "sentences")
        itemToEdit.questions[QUESTIONS[action.question]] = itemToEdit[
          action.typeOfWord
        ].join(" ");
      else
        itemToEdit.questions[QUESTIONS[action.question]] = itemToEdit[
          action.typeOfWord
        ].join(", ");

      return {
        ...state,
        summaryOfChosenItems: {
          ...state.summaryOfChosenItems,
          [state.activeSummaryItemId]: itemToEdit,
        },
      };
    }

    case ADD_SUMMARY_ITEM:
      return {
        ...state,
        activeSummaryItemId: action.id,
        summaryOfChosenItems: {
          ...state.summaryOfChosenItems,
          [action.id]: {
            id: action.id,
            name: "",
            cards: [],
            feelingsWords: [],
            feelingsYouMissWords: [],
            needs: [],
            needsBehindAFeeling: [],
            sentences: [],
            questions: {},
            itemSaved: false,
          },
        },
      };

    case DELETE_SUMMARY_ITEM:
      const newSummaryItems = { ...state.summaryOfChosenItems };
      delete newSummaryItems[action.id];
      return {
        ...state,
        activeSummaryItemId:
          state.activeSummaryItemId === action.id
            ? ""
            : state.activeSummaryItemId,
        summaryOfChosenItems: newSummaryItems,
      };

    case ADD_ANSWER_TO_A_QUESTION:
      const newSummaryItem = { ...state.summaryOfChosenItems[action.id] };
      newSummaryItem.questions[action.question] = action.answer;

      if (!action.listMode && action.wordType) {
        newSummaryItem[action.wordType] = [];
      }

      return {
        ...state,
        summaryOfChosenItems: {
          ...state.summaryOfChosenItems,
          [action.id]: newSummaryItem,
        },
      };

    case LOAD_SUMMARY_ITEMS:
      return {
        ...state,
        initialLoadFinished: true,
        summaryOfChosenItems: action.items,
        numOfItems: Object.keys(action.items).length,
      };

    default:
      return state;
  }
};

export default reducer;
