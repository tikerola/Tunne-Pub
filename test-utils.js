import React from "react";
import { NavigationContext } from "@react-navigation/native";
import { render } from "@testing-library/react-native";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import CARD_DATA from "./data/cards";
import NEEDS_DATA from "./data/needs";
import { SENTENCES_DATA } from "./data/sentences";
import WORD_DATA from "./data/words";

const INITIAL_STATE = {
  cards: {
    // cards: CARD_DATA,
    // words: WORD_DATA,
    // needs: NEEDS_DATA,
    // sentences: SENTENCES_DATA,
    activeSummaryItemId: "123",
    summaryOfChosenItems: {
      123: {
        id: "123",
        name: "",
        cards: [],
        feelingsWords: [],
        feelingsYouMissWords: [],
        needs: ["hei"],
        needsBehindAFeeling: [],
        sentences: [],
        questions: {},
        itemSaved: true,
      },
    },
    screenNumber: 1,
    numOfItems: 0,
    initialLoadFinished: false,
  },
};

const mockStore = configureStore([]);
const store = mockStore(INITIAL_STATE);

const navContext = {
  isFocused: () => true,
  // addListener returns an unscubscribe function.
  addListener: jest.fn(() => jest.fn()),
};

const reduxRender = (ui, options, storeData = store) =>
  render(ui, {
    wrapper: ({ children }) => (
      <Provider store={storeData}>
        <NavigationContext.Provider value={navContext}>
          {children}
        </NavigationContext.Provider>
      </Provider>
    ),
    ...options,
  });

// re-export everything for convenience
export * from "@testing-library/react-native";

// override render method
export { reduxRender as render, store };
