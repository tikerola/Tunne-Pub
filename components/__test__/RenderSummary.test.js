import React from "react";
import { QUESTIONS } from "../../data/questions";
import { render } from "../../test-utils";
import { RenderSummary } from "../RenderSummary";
import configureStore from "redux-mock-store";
import Card from "../../models/card";
import CARD_DATA from "../../data/cards";
import { fireEvent, waitFor } from "@testing-library/react-native";
import { deleteSummaryItem } from "../../store/actions/cards";

describe("<RenderSummary />", () => {
  const allCards = [
    CARD_DATA[0],
    CARD_DATA[1],
    CARD_DATA[2],
    CARD_DATA[3],
    CARD_DATA[4],
    CARD_DATA[5],
  ];
  const data = [CARD_DATA[0], CARD_DATA[1], CARD_DATA[2]];

  const INITIAL_STATE = {
    cards: {
      cards: allCards,
      activeSummaryItemId: "123",
      summaryOfChosenItems: {
        123: {
          id: "123",
          name: "my name",
          cards: data,
          feelingsWords: ["pahoillaan", "yskä"],
          feelingsYouMissWords: [],
          needs: ["hei"],
          needsBehindAFeeling: [],
          sentences: [],
          questions: {
            [QUESTIONS[7]]: "niinpä niin",
            [QUESTIONS[8]]: "pahoillaan, yskä",
            [QUESTIONS[9]]: "diipadaapa",
          },
          itemSaved: true,
        },
        234: {
          id: "234",
          name: "",
          cards: data,
          feelingsWords: ["pahoillaan", "yskä"],
          feelingsYouMissWords: [],
          needs: ["hei"],
          needsBehindAFeeling: [],
          sentences: [],
          questions: {
            [QUESTIONS[7]]: "niinpä niin",
            [QUESTIONS[8]]: "pahoillaan, yskä",
            [QUESTIONS[9]]: "diipadaapa",
          },
          itemSaved: true,
        },
      },
      screenNumber: 1,
      numOfItems: 0,
      initialLoadFinished: false,
    },
  };

  const navigation = {
    navigate: jest.fn(),
    goBack: jest.fn(),
  };

  const dispatch = jest.fn();

  const mockStore = configureStore([]);
  const store = mockStore(INITIAL_STATE);

  it("should match snapshot", () => {
    expect(
      render(
        <RenderSummary
          navigation={navigation}
          item={INITIAL_STATE.cards.summaryOfChosenItems["123"]}
          dispatch={dispatch}
        />,
        undefined,
        store
      )
    ).toMatchSnapshot();
  });

  it("should be able to be edited if marked as editable", () => {
    const component = render(
      <RenderSummary
        navigation={navigation}
        item={INITIAL_STATE.cards.summaryOfChosenItems["123"]}
        dispatch={dispatch}
        enableEdit={true}
      />,
      undefined,
      store
    );

    const editButton = component.queryByText("Muokkaa");
    fireEvent.press(editButton);

    expect(dispatch).toHaveBeenCalled();
    expect(navigation.navigate).toHaveBeenCalledWith("MainScreen", {
      name: "my name",
    });
  });

  it("should be able to delete if marked as editable", async () => {
    const component = render(
      <RenderSummary
        navigation={navigation}
        item={INITIAL_STATE.cards.summaryOfChosenItems["123"]}
        dispatch={dispatch}
        enableEdit={true}
      />,
      undefined,
      store
    );

    const deleteButton = component.queryByText("Poista");
    fireEvent.press(deleteButton);

    expect(
      component.queryByText("Haluatko varmasti poistaa istunnon?")
    ).toBeTruthy();

    const confirmButton = component.queryByText("Kyllä");
    fireEvent.press(confirmButton);
    expect(navigation.goBack).toHaveBeenCalled();

    expect(dispatch).toHaveBeenCalledWith(deleteSummaryItem("123"));
  });
});
