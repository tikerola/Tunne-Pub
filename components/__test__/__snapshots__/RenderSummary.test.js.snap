// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RenderSummary /> should match snapshot 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "paddingBottom": 15,
      "paddingTop": 10,
      "width": "100%",
    }
  }
>
  <View>
    <RCTScrollView
      data={
        Array [
          Card {
            "feeling": "happy",
            "filename": 1,
            "id": "1",
          },
          Card {
            "feeling": "sad",
            "filename": 1,
            "id": "2",
          },
          Card {
            "feeling": "worried",
            "filename": 1,
            "id": "3",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 140,
                "marginLeft": 5,
                "marginRight": 5,
                "width": 96,
              }
            }
          />
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 140,
                "marginLeft": 5,
                "marginRight": 5,
                "width": 96,
              }
            }
          />
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 140,
                "marginLeft": 5,
                "marginRight": 5,
                "width": 96,
              }
            }
          />
        </View>
      </View>
    </RCTScrollView>
    <View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0.5)",
            "display": "flex",
            "flexDirection": "row",
            "marginTop": 10,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#bbb",
              "fontFamily": "Roboto-Light",
              "fontSize": 18,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#666",
                "fontWeight": "bold",
              }
            }
          >
            Mikä tarve sinulla on juuri nyt?
            : 
          </Text>
          niinpä niin
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0.5)",
            "display": "flex",
            "flexDirection": "row",
            "marginTop": 10,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#bbb",
              "fontFamily": "Roboto-Light",
              "fontSize": 18,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#666",
                "fontWeight": "bold",
              }
            }
          >
            Mitä lausetta pyörität päässäsi juuri nyt? Mikä uskomus sinulla on? Mitä se sinulle merkitsee?
            : 
          </Text>
          pahoillaan, yskä
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0.5)",
            "display": "flex",
            "flexDirection": "row",
            "marginTop": 10,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#bbb",
              "fontFamily": "Roboto-Light",
              "fontSize": 18,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#666",
                "fontWeight": "bold",
              }
            }
          >
            Mitä sinun nyt pitäisi tehdä jotta voisit hyvin?
            : 
          </Text>
          diipadaapa
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
